clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/deploy/asg/AsgConfigHelper.java

```diff
@@ -332,6 +332,35 @@ private static String createDefaultSuffix() {
         .collect(Collectors.toList());
   }
 
+  protected static List<AmazonBlockDevice> convertBlockDevices(
+      List<com.amazonaws.services.ec2.model.BlockDeviceMapping> blockDeviceMappings) {
+    blockDeviceMappings.forEach(b -> log.info(b.getDeviceName()));
+    return blockDeviceMappings.stream()
+        .map(
+            bdm -> {
+              AmazonBlockDevice amzBd =
+                  new AmazonBlockDevice.Builder()
+                      .deviceName(bdm.getDeviceName())
+                      .virtualName(bdm.getVirtualName())
+                      .build();
+
+              if (bdm.getEbs() != null) {
+                final EbsBlockDevice ebs = bdm.getEbs();
```

Comment:
Looks like AmazonBlockDevice has a new field called kmsKeyId. We may need to do a null check/set setKmsKeyId here? EbsBlockDevice has getKmsKeyId(), however Ebs does not so it would be just needed for this method

Reply:
Added